vpn/client.go

```diff
@@ -83,18 +85,30 @@ func (c *Client) Start(ctx context.Context) error {
 	}
 	c.conn = conn
 
-	mux, err := newMuxerFromOptions(conn, c.Opts, c.tunnel)
-	if err != nil {
-		return err
-	}
-	err = mux.Handshake(ctx)
-	if err != nil {
-		return err
+	if c.mux == nil {
+		muxFactory := c.muxerFactory()
+		mux, err := muxFactory(conn, c.Opts, c.tunnel)
+		if err != nil {
+			return err
+		}
+		err = mux.Handshake(ctx)
+		if err != nil {
+			return err
+		}
+		c.mux = mux
```

Comment:
I am not super convinced by the fact that calling `Start` multiple times creates a new connection each time, however there is just a single `mux` factory that is created once. We should have a design discussion about this topic. My instinct is that either the whole of start is idempotent after the first call, or none of start is idempotent, but I am very surprised by a situation where just part of it are idempotent.

Reply:
incorporated these changes after our discussion today (I think I like #approach1 more, it makes more explicit that start is only supposed to be called once).